### DISCUSSION TITLE
Tasmota battery remote controller?

### DISCUSSION THREAD

**Discussion by drgkt** [arendst/Tasmota#12598] [Q&A] [answered]
I was surprised I could not find a battery operated programmable multi-button remote to on / off my plugs locally when my phone is not around.
One like this will control up to 20 plugs.
![20210708_202001](https://user-images.githubusercontent.com/32744663/124964855-2a6d8180-e02a-11eb-812b-68307f79170a.jpg)

Any reason why no one makes those yet?

**Comment by sfromis** [comment, marked as answer]
Tasmota primarily runs on ESP8266 which is a bad match for battery operation due to the power consumption. If you want wireless remote control with Tasmota, you can get a Zigbee button and have a Tasmota-flashed coordinator. A lot of the time, you can also get old-fashioned RF 433 or IR remotes to work with Tasmota on a suitable bridge device.

**Comment by drgkt** [comment]
Zigbee: do I need one button for each device? If yes, not practical.
RF 433: You mean I can use this pictured multibutton remote to control (on / off) up to 20 tasmota plugs with a bridge?
If yes, I need more info since I know nothing about. I have the RF transmitter, how to get the bridge and what I need to do?
Thanks

**Reply by sfromis** [reply]
Zigbee remotes with multiple buttons exist, saw one up to 20. https://zigbee.blakadder.com/remotes.html

RF 433 can be done with the [Sonoff RF Bridge](https://tasmota.github.io/docs/devices/Sonoff-RF-Bridge-433/) (and similar devices). As RF 433 is an outdated technology without clear standards, there is a risk of a particular device not just so being supported.

**Comment by drgkt** [comment]
So zigbee is the recommended choice. And where to get the coordinator?

**Reply by sfromis** [reply]
https://zigbee.blakadder.com/coordinators.html

An "easy" one is the Sonoff ZbBridge, a "complete" device flashable with Tasmota.

**Comment by drgkt** [comment]
Ok then. Summarizing, need
a. https://zigbee.blakadder.com/DIYRuZ_FreePad.html (The only one with 20 buttons on that page)
and
b. https://zigbee.blakadder.com/Sonoff_ZBBridge.html

Correct?   Note that only local control is desired, so no need for MQTT

**Reply by sfromis** [reply]
"Local" may be a somewhat ambiguous word. Understood as "in my house", I see MQTT as the best option for controlling other Tasmota devices on my WLAN, preferable to, say, HTTP. I've not used that 20-button device, but my Sonoff ZbBridge runs fine for a few other Zigbee devices.

**Reply by barbudor** [reply]
@drgkt 
Using Tasmota is meaningful if you go to some level of home automation integration.
This means you need to have a minimum infrastructure with a MQTT broker, dashboards, automation, ... A RaspberryPI running MQTT and Node-Red or OpenHAB for example is by-default purely local. This has nothing to do with local or non-local. This has to do with what you want to achieve.

If what you want is "just" to turn on and off a set plugs and lights from a remote control, you don't need Tasmota and you can get a bunch of Zigbee devices, pair them together and you are done. There's a ton of such devices, even from Lidl.

You can use Tasmota without a MQTT broker, but its just not the way it was intended to.

**Comment by drgkt** [comment]
I realize that tasmota has much more to offer than I use it for.
I first used WEMO, then eWeLink devices, but they do not have http web interface and I needed that to use with tasker. That is when I came accross Tasmota. So I got a couple preloaded (Athom), gave them fixed IP and I use them locally, meaning within my house via  http or TasmotaRemota app. I do not have a need to control those from anywhere in the world, so (as I understand it) no need for MQTT. That is what I mean local, and it is for those I was wondering about button remote control (in the absence of pc or phone).
I still use the eWeLink devices where I want to have control from anywhere in the world, but eventually I will tasmotize them, and for those cases that I will need worrld wide control, yes I will integrate MQTT, again if I understand it correctly.

By the way, if I set a timer on a tasmota device, even a sunrize / sunset event, then move them in an area with no access to wifi, will they still work?

**Reply by sfromis** [reply]
The question of "from anywhere in the world" is simply not related to using MQTT or not. I consider MQTT as being *better* for local operation (in my house) than getting tricky with safe remote operation (MQTT-TLS), where I'd be inclined to set up a VPN tunnel instead.

Tasmota devices need a real time source, which by default is accessing NTP servers via wifi+Internet. You can also have a local NTP server, or add a real time chip to a device running Tasmota.
